gtsam/hybrid/tests/testGaussianMixtureFactor.cpp

```diff
@@ -191,8 +197,422 @@ TEST(GaussianMixtureFactor, Error) {
   DiscreteValues discreteValues;
   discreteValues[m1.first] = 1;
   EXPECT_DOUBLES_EQUAL(
-      4.0, mixtureFactor.error({continuousValues, discreteValues}),
-      1e-9);
+      4.0, mixtureFactor.error({continuousValues, discreteValues}), 1e-9);
+}
+
+/* ************************************************************************* */
+/**
+ * Test a simple Gaussian Mixture Model represented as P(m)P(z|m)
+ * where m is a discrete variable and z is a continuous variable.
+ * m is binary and depending on m, we have 2 different means
+ * μ1 and μ2 for the Gaussian distribution around which we sample z.
+ *
+ * The resulting factor graph should eliminate to a Bayes net
+ * which represents a sigmoid function.
+ */
+TEST(GaussianMixtureFactor, GaussianMixtureModel) {
+  double mu0 = 1.0, mu1 = 3.0;
+  double sigma = 2.0;
+  auto model = noiseModel::Isotropic::Sigma(1, sigma);
+
+  DiscreteKey m(M(0), 2);
+  Key z = Z(0);
+
+  auto c0 = make_shared<GaussianConditional>(z, Vector1(mu0), I_1x1, model),
+       c1 = make_shared<GaussianConditional>(z, Vector1(mu1), I_1x1, model);
+
+  auto gm = new GaussianMixture({z}, {}, {m}, {c0, c1});
+  auto mixing = new DiscreteConditional(m, "0.5/0.5");
+
+  HybridBayesNet hbn;
+  hbn.emplace_back(gm);
+  hbn.emplace_back(mixing);
+
+  // The result should be a sigmoid.
+  // So should be m = 0.5 at z=3.0 - 1.0=2.0
```

Comment:
Gravy: derive sigmoid, add a lambda, and compare at 3 different places.

Since $P(z,m) = P(z|m)P(m)$, and $P(m)=50/50$, we have 

$$
P(m|z) \propto P(z|m) \propto exp(-|z-\mu_m|^2/4)
$$

$$
P(0|z) = \frac{exp(-|z-\mu_0|^2/4)}{exp(-|z-\mu_0|^2/4)+exp(-|z-\mu_1|^2/4)}
$$

$$
P(0|z) = \frac{1}{1+exp(|z-\mu_0|^2/4-|z-\mu_1|^2/4)}
$$

<img width="352" alt="image" src="https://github.com/user-attachments/assets/8d66e1ce-1cb1-4e52-911e-3ba798275bee">

Reply:
Still only compared at special case 50/50 point. Could be wrong in other places.